### ISSUE TITLE
Basic Word add-in 'Hello world': the web app rejects the taskpane.css as having type 'text/html' - not always,  see my report

### ISSUE THREAD

**Issue by rudifa** [OfficeDev/office-js#2948]
**If this is not the right place to report this issue please redirect me**

I observed a **frustrating problem** while working with the **basic Word add-in** 
-  I created the add-in project per tutorial [Build your first Word task pane add-in](https://learn.microsoft.com/en-us/office/dev/add-ins/quickstarts/word-quickstart?tabs=yeomangenerator)
- I am testing in the **online MS Word app** served from [office.com](https://www.office.com/?auth=1)
- I am working on a **Mac** 12.6 Monterey, using the VSCode editor
- The same problem occurs in any of my browsers (Edge, Chrome, Safari, Brave)

The problem is that in certain reloads (see below) the **styling** defined in `src/taskpane.css` is **not applied** so that the UI is jumbled up. Indeed, the Console output shows this diagnostic:

```
taskpane.html:1     
       Refused to apply style from 'https://localhost:3000/5c1b8bbd1b284c38b5de.css' because its MIME type ('text/html') is not a supported stylesheet MIME type, and strict MIME checking is enabled.
```

**To reproduce the bug:**

1. start the MS Word online app in the browser (log in or sign in) and open a document
2. start the dev server as indicated in the tutorial (having obtained your document's url from the app's `Share - Copy link`

`npm run start:web -- --document 'https://1drv.ms/w/s!AoxMqCIhJWzTgQs63z8Fb3AMQzx-?e=wD8Phh'`

3. open the Taskpane (its UI should look good) and click on the `Run` button to insert `Hello World` text into the document
4. in the VScode editor change the text, e.g. `Hello World 1`; save
5. the dev server reloads the add-in
6. click on `Run` to confirm that the modified text gets inserted
7. already at this stage, or at the latest after you close + reopen the Taskpane, the UI will be jumbled
8. workaround: modify the file src/taskpane.html, e.g add a space in one of the html comments, save
9. the dev server reloads the add-in
10. close + reopen the Taskpane: the UI will be restored to its normal good state

I have repeated the cycle 4 - 10 many times, always breaking and then fixing the UI styling.

**Questions to the community:**

Can anyone reproduce this bug?
Does it occur also on the desktop MS Word app (I have no access to a Windows PC right now)?
Can anyone diagnose this and propose a solution?

**Comment by rudifa** [comment]
1. Searching stackoverflow for `because its MIME type ('text/html') is not a supported stylesheet MIME type`  produces several answers along these lines:
```
It happens when you set an incorrect URL to the file or when your server isn't configured properly.
In the result, the browser DOESN'T get the stylesheet, but it gets some HTML with 404 status and 
with the "Content-Type" header. Since the browser gets something from the server, it doesn't tell you 
there is no reply, but it tells you the MIME type of the file is incorrect. 
```
If this be the case here, perhaps the webpack.config.js should be tweaked, but I don't know how.

2. I understand that the webpack-dev-server keeps its intermediate files (the equivalent of dist/* files in the build) in memory, so we can't see them.

The alternating bug pattern that I observe (a change in taskpane.html fixes it, a change in taskpane.js breaks it) suggests that in the broken case the server does not have the .css file in its in-memory website, and instead of regenerating it, returns a not-found html file.

HTH

**Comment by ElizabethSamuel-MSFT** [comment]
@rudifa Thanks for bringing this to our attention. It looks like we might need to update the template.

In the meantime, can you try the following?

1. Install additional loaders by running the following command.
    ```
    npm install --save-dev style-loader css-loader
    ```
    Example screenshot (note that it may not match your environment):
    ![image](https://user-images.githubusercontent.com/28541963/200415944-9f3f0646-51df-4d70-aba6-5074ee7fa5d0.png)

2. Update **taskpane.js** by inserting the following code before the Office.onReady statement.
    ```
    require("!style-loader!css-loader!./taskpane.css");
    ```
    Excerpt:
    ![image](https://user-images.githubusercontent.com/28541963/200416725-e9bdc06d-6a5e-468d-b4ce-53ccb1e70ae9.png)

Let us know if this helps.

Thanks.

**Comment by RuizhiSunMS** [comment]
Create ADO#7093555 to track the change in template. Sorry for unable to provide exact schedule but will update here as soon as the progress is made.
Thanks.

**Comment by rudifa** [comment]
Hi @ElizabethSamuel-MSFT @RuizhiSunMS @xiruatms 

Thank you for the rapid response.

Indeed, the suggested modifications fix the problem - now the automatic reload of the add-in in the Word web app after a change in `taskpane.js` does not break the styling. Tested on my Mac 12.6, in Edge and in Chrome browsers.

```
My-Office-Add-in % git diff -U1 package.json   
diff --git a/package.json b/package.json
index 97644d9..d84a29a 100644
--- a/package.json
+++ b/package.json
@@ -40,2 +40,3 @@
     "copy-webpack-plugin": "^9.0.1",
+    "css-loader": "^6.7.1",
     "eslint-plugin-office-addins": "^2.1.0",
@@ -53,2 +54,3 @@
     "source-map-loader": "^3.0.0",
+    "style-loader": "^3.3.1",
     "ts-loader": "^9.3.1",
``` 
```
My-Office-Add-in % git diff -U2 src/taskpane/taskpane.js
diff --git a/src/taskpane/taskpane.js b/src/taskpane/taskpane.js
index 4e5506d..61e0da9 100644
--- a/src/taskpane/taskpane.js
+++ b/src/taskpane/taskpane.js
@@ -6,4 +6,6 @@
 /* global document, Office, Word */
 
+require("!style-loader!css-loader!./taskpane.css");
+
 Office.onReady((info) => {
   if (info.host === Office.HostType.Word) {
```

**Comment by ElizabethSamuel-MSFT** [comment]
@rudifa Thanks for letting us know and so happy to hear it helped.

**Comment by rudifa** [comment]
@ElizabethSamuel-MSFT 
I tested some more your fix,  on another add-in.
It works as expected (no more lost styling) on Edge, Brave, Safari and Chrome (my default browser).

However, when I tested it for the first time on Firefox 106.0.5 (MAcoS), the add-in was rejected.
Obviously, this is not a blocking problem for me.
However, i will open a separate issue.

**Comment by ElizabethSamuel-MSFT** [comment]
@rudifa Yes, please do create a new issue with the details then we'll dig in there. Thanks.

**Comment by rudifa** [comment]
Done.

**Comment by rudifa** [comment]
@ElizabethSamuel-MSFT : some concluding remarks

1. FYI, here  is my [add-in project](https://github.com/rudifa/Office-Add-in-2) on which I discovered the problem of the vanishing styling.

2. I fixed my project according to your suggestions, (installed the 2 loaders and required the `taskpane.css` file. I also removed from `taskpane.html `the clause `<link rel="stylesheet" type="text/css" href="taskpane.css"  />`, and it works for me.

3. I looked into Webpack documentation and saw that they have two methods for dealing with .css and other non-js files, 
```
[Configuration]: Specify them in your webpack.config.js file.
[Inline]: Specify them explicitly in each import statement.
```
Your suggestions correspond to the inline method. 
I experimented looking for an equivalent configuration, but I failed to find one.
